pages/directory/index.tsx

```diff
@@ -1,13 +1,72 @@
-import { Box, Container, List, ListItem, Typography } from '@mui/material';
+import { Box, Container, CircularProgress, List, ListItem, Typography } from '@mui/material';
+import axios from 'axios';
 import { NextPage } from 'next';
 import React from 'react';
 
 import createAxiosInstance from 'client/lib/axios';
 
 //TODO: Type users once sequelize-typescript is added
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const Directory: NextPage<{ users: any[]; error: string }> = props => {
-  const { users, error } = props;
+const Directory: NextPage<{ users: any[]; totalPages: number; error: string }> = props => {
+  const { users, totalPages, error } = props;
+  const lastCardRef = React.useRef(null);
+  const [userData, setUserData] = React.useState(users);
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [totalPageNum, setTotalPageNum] = React.useState(totalPages);
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  async function fetchUsers(page: number) {
+    try {
+      const response = await axios.get(`/api/users?page=${page}`);
+      return {
+        page: response.data.page,
+        total: response.data.totalPages,
+        users: response.data.users,
+      };
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  React.useEffect(() => {
+    const height = getElementHeight(document.querySelector('#last-card'));
+
+    function getElementHeight(el: HTMLElement) {
+      const rect = el.getBoundingClientRect();
+      const { marginTop, marginBottom } = getComputedStyle(el);
+
+      return rect.height + parseFloat(marginTop) + parseFloat(marginBottom);
+    }
+
+    const intersectionOptions = {
+      // root default = document's viewport
+      rootMargin: `${height}px`,
+      threshold: 0,
+    };
+
+    const lastCardObserver = new IntersectionObserver(async ([lastCard]) => {
+      // guard clause, should intersect only once
+      if (!lastCard.isIntersecting) return;
```

Comment:
nit: add linebreak after `return;` to make it more readable by adding a separation in logic.